Automattic/wp-calypso
# Shuffle site goal values

We have a hunch that people's choices for their site goals are being influenced by the order of the goals listed on the new site segmenting screen. In this PR, I've implemented an A/B test that will allows us to see if this is true. 50% of visitors will all see the options in the same order vs the other 50% that will each see the options listed in different order. 

The options only needed to be randomized once so I used a combination of the redux state and local storage to ensure that the results stay the same after the first time they're loaded. 

## Before
![image](https://user-images.githubusercontent.com/6981253/36223708-3d92e250-1193-11e8-9285-602762070c06.png)

## After
![image](https://user-images.githubusercontent.com/6981253/36223686-2eb3b304-1193-11e8-8ca6-566d70f03e36.png)

## Testing
- Visit `/start`
- Make sure you're in the `variant` group for `siteGoalsShuffle`
- Clear `setSiteGoalsArray` from your local storage
- Refresh your screen to see the primary goals sorted in a new order
- Refresh your screen again to make sure they stay the same
- Select an option and then continue to the next step
- Return back to the first step to make sure the goals are sorted in the same order as when you left. 

@taggon can you take a look?
